src/main/java/org/jenkinsci/plugins/sonargerrit/config/InspectionConfig.java

```diff
@@ -160,8 +230,64 @@ public FormValidation doCheckServerURL(@QueryParameter String value) {
 
         }
 
+        /**
+         * Is only called once, filtering is done in Frontend by a Combo Box
+         *
+         * @param value component to search for
+         * @param sonarInstallationName  specifies the SonarQube server where the components are fetched from
+         * @param analysisType only for Analysis Type.PULL_REQUEST the components are inserted into the model's list of components
+         * @return model containing a list of components matching the given component name
+         * @throws AbortException if Sonar installation cannot be found
+         */
+        @SuppressWarnings("unused")
+        public ComboBoxModel doFillComponentItems(@QueryParameter String value, @QueryParameter String sonarInstallationName,
+                @QueryParameter AnalysisType analysisType) throws AbortException {
+            if (analysisType == AnalysisType.PULL_REQUEST) {
+                SonarClient sonarClient = SonarUtil.getSonarClient(sonarInstallationName);
+                String componentKey = SonarUtil.isolateComponentKey(value);
+                ComponentSearchResult componentSearchResult = sonarClient.fetchComponent(componentKey);
+                return new ComboBoxModel(componentSearchResult.getComponents().stream()
+                        .map(c -> c.getName() + " (" + c.getKey() + ")")
+                        .collect(Collectors.toList()));
+            } else {
+                return new ComboBoxModel();
+            }
+        }
+
+        @SuppressWarnings("unused")
+        public FormValidation doCheckComponent(@QueryParameter String value, @QueryParameter String sonarInstallationName,
+                @QueryParameter AnalysisType analysisType) throws AbortException {
+            if (analysisType == AnalysisType.PULL_REQUEST) {
+                SonarClient sonarClient = SonarUtil.getSonarClient(sonarInstallationName);
+                String componentKey = SonarUtil.isolateComponentKey(value);
+                ComponentSearchResult componentSearchResult = sonarClient.fetchComponent(componentKey);
+
+                if (componentSearchResult.getPaging().getTotal() == 1) {
```

Comment:
total is an Integer, could this trigger an NPE?

Reply:
was never null, if so I think I will not get a result at all